src/modules/auth/clients/ClientCreateContainer.js

```diff
@@ -280,143 +188,224 @@ const ClientCreateContainer = (props) => {
-											case 'code_challenge_methods': return(<RadioInput key={key} name={key} register={register} valueList={value["items"]["enum"]} disabled={disabled} labelName={t('ClientCreateContainer|Code challenge methods')}/>)
-											case 'application_type': return(selectedTemplate === "Custom" && <SelectInput key={key} name={key} register={register} valueList={value["enum"]} disabled={disabled} labelName={t('ClientCreateContainer|Application type')}/>)
-											case 'token_endpoint_auth_method': return(selectedTemplate === "Custom" && <SelectInput key={key} name={key} register={register} valueList={value["enum"]} disabled={disabled} labelName={t('ClientCreateContainer|Token endpoint authentication method')}/>)
-											default: return(<div key={key}>{t('ClientCreateContainer|Unknown item')}: "{key}"</div>)
-										}
-									}
-								})}
-							</CardBody>
-							<CardFooter>
+		<Container className="client-wrapper">
+			<Form className="client-cards-wrapper" onSubmit={(client == undefined) ? handleSubmit(onSubmitNewClient) : handleSubmit(onSubmitEditClient)}>
+				<div className="client-main-info">
+					<Card>
+						<CardHeader className="border-bottom">
+							<div className="card-header-title">
+								<i className="cil-layers pr-2"></i>
 								{((client != undefined) && (editClient == true)) ?
-									<ButtonGroup>
-										<ButtonWithAuthz
-											title={t("ClientListContainer|Save")}
-											color="primary"
-											type="submit"
-											disabled={isSubmitting}
-											resource={resource}
-											resources={resources}
-										>
-											{t("ClientListContainer|Save")}
-										</ButtonWithAuthz>
-										<ButtonWithAuthz
-											outline
-											title={t("Cancel")}
-											color="primary"
-											type="submit"
-											disabled={isSubmitting}
-											resource={resource}
-											resources={resources}
-											onClick={() => props.history.push(`/auth/clients/${client_id}`)}
-										>
-											{t("Cancel")}
-										</ButtonWithAuthz>
-									</ButtonGroup>
-
+									t("ClientCreateContainer|Edit client")
 								:
+									t("ClientCreateContainer|Create new client")
+								}
+							</div>
+						</CardHeader>
+
+						<CardBody>
+							<TextInput
+								name="client_name"
+								register={register}
+								required={true}
+								disabled={disabled}
+								labelName={`${t("ClientCreateContainer|Client name")}*`}
+							/>
+							<URiInput
+								name="redirect_uris_main"
+								templateName="redirect_uris"
+								invalid={errors?.redirect_uris_main && true}
+								disabled={disabled}
+								errors={errors}
+								append={append}
+								remove={remove}
+								fields={fields}
+								register={register}
+								reg={regRedirectUrisMain}
+								labelName={`${t("ClientCreateContainer|Redirect URIs")}*`}
+							/>
+							{metaData["properties"] && Object.entries(metaData["properties"]).map(([key, value]) => {
```

Comment:
@Pe5h4 I don't quite understand how I can use the find method here. I need to return an input, but the find method won't do that

Reply:
@Aringoaway Oh, I see, but I still dont understand, why you need to use iteration (it can be expensive), when trying to find one specific key and return its content


Can be the content of the `redirect_uri_validation_method` obtained as:

```
{metaData["properties"] && metaData["properties"]["redirect_uri_validation_method"] &&
<RadioInput
	key={key}
	name={key}
	register={register}
	valueList={metaData["properties"]["redirect_uri_validation_method"]["enum"]}
	disabled={disabled}
	labelName={t('ClientCreateContainer|Redirect URI validation method')}
	editing={(client != undefined)}
/>}
```